### DISCUSSION TITLE
[V5] Argument of type 'string' is not assignable to parameter of type '"api"'.ts(2345)

### DISCUSSION THREAD

**Discussion by remidosol** [adonisjs/core#1604] [Help] [answered]
Hello there, I've a issue about that happens after _"node ace invoke @adonisjs/auth"_ command.

I got this error in **"app/Middleware/Auth.ts"** file's **27th** line: 
**"Argument of type 'string' is not assignable to parameter of type '"api"'.ts(2345)"**

This is the function that I've got error: 
```typescript
protected async authenticate (auth: HttpContextContract['auth'], guards: string[]) {
    for (let guard of guards) {
      if (await auth.use(guard).check()) {  //! Warning is appearing on the 'guard' variable on this line.
        auth.defaultGuard = guard
        return true
      }
    }
```

And here it is my **contracts/auth.ts**

```typescript
interface ProvidersList {
    user: {
      implementation: LucidProviderContract<typeof User>,
      config: LucidProviderConfig<typeof User>,
    },
  }
  interface GuardsList {
    api: {
      implementation: OATGuardContract<'user', 'api'>,
      config: OATGuardConfig<'user'>,
    },
  }
```

Could you help me, please?

**Comment by thetutlage** [comment, marked as answer]
Seems weird. Can you share when did you installed the `@adonisjs/auth` package. Coz the middleware is using an outdated template (which shouldn't be the case for a new app).

Anyways, I recommend replacing the contents of `app/Middleware/Auth` with this file https://github.com/adonisjs/auth/blob/develop/templates/middleware/Auth.txt

**Reply by remidosol** [reply]
in package.js file: "@adonisjs/auth": "^4.2.4"
I installed today.

**Reply by remidosol** [reply]
Thanks.
